### PR TITLE
Display full error message from uncaught exceptions

### LINKED ISSUE #737: Only "Uncaught" shown for some errors
Running https://dartpad.dartlang.org/bad57c341ff1dbbc8e7464cd458b124b shows:

> Uncaught

instead of something like

> CastError: Casting value of type JSArray to incompatible type num
#705 related or the same?

### PR DESCRIPTION
**Pull request by jcollins-g**
Fixes #737, hopefully for real this time.

In simplifying my fix I had accidentally removed the part that was essential because I had tested it on a bad example.   errorObject sometimes contains what's necessary to fully describe the error, and I've verified it with the case mentioned in #737 that's still valid in strong mode:

screenshot from 2018-05-31 13-42-42

### CODE REVIEW

**Comment by jcollins-g** [comment]
putting this on hold for a moment.  It seems sometimes errorObject is needed, and sometimes it duplicates information already displayed.

**Comment by jcollins-g** [comment]
Latest patch makes this work properly too, without doubling up on part of the error message.  (The example is only valid in Dart 1 land, but want to be sure that any other exceptions of the same type are handled).

screenshot from 2018-05-31 13-59-14

File: lib/services/execution_iframe.dart [hunk 1]
```diff
@@ -55,9 +55,16 @@ function dartPrint(message) {
 ''';
 
     final String exceptionHandler = '''
-window.onerror = function(message, url, lineNumber) {
+window.onerror = function(message, url, lineNumber, columnNumber, errorObject) {
+  out = '';
```
Comment (devoncarew) [inline lib/services/execution_iframe.dart]: `var out = '';`?

**Comment by jcollins-g** [comment]
PTAL: the try/catch method of handling exceptions is more portable and appears to work not only on Chrome, but on Firefox/Safari as well.

I've also reformatted how they appear to be closer to what the VM outputs for an exception.

**Comment by jcollins-g** [comment]
A version of the Dart 2.0 supporting dartpad with this patched in is now available at https://dart2-test-dot-dart-pad.appspot.com/.

**Pull request merged**